processing/src/main/java/io/druid/query/filter/InDimFilter.java

```diff
@@ -117,6 +121,46 @@ public ExtractionFn getExtractionFn()
   @Override
   public DimFilter optimize()
   {
+    InDimFilter inFilter = optimizeLookup();
+    if (inFilter.values.size() == 1) {
```

Comment:
I believe there is a check in the constructor right now
Preconditions.checkArgument(values != null && !values.isEmpty(), "values can not be null or empty");